### ISSUE TITLE
Challenges to Add new Search Engine?

### ISSUE THREAD

**Issue by lonnietc** [ik9999/popupsearch#16]
Hi,

Can you please give me an idea as to what is required in order to add a new search engine to the list of available engines?

I am trying to determine what files may need to be created, and/or changed.

Thanks

**Comment by ik9999** [comment]
https://github.com/ik9999/popupsearch/tree/master/src/popup/search
Add a new file here (see files for google and searx for an example of a structure).

Then call it here:

https://github.com/ik9999/popupsearch/blob/master/src/popup/store/modules/SearchResults.js#L88

**Comment by lonnietc** [comment]
Thanks for the reply on this and I will dig into it more.

Actually, and even though SearX is a wonderful meta search engine, I was thinking to build out a metasearch engine core selection from within popupsearch that would call a number of selectable engines and aggregate the results so that you would not have to call SearX, or could actually call it along with the others as well just to add a bit more local functionality to the extension.

The idea was to have about a few, or more selectable engines that could be available for the local metasearch engine.

On a side question, does popupsearch filter out things to try and keep the search anonymous as possible although there are, of course, limitations to what anonymous returns can be done from within the browser but that would be very nice as well to have locally too.

Anyway, it was just a thought.
Thanks again.

**Comment by ik9999** [comment]
> does popupsearch filter out things to try and keep the search anonymous as possible although there are

No, it really doesnt. When you search in google, it still sends google the cookies from the browser. Not sure if its possible to prevent that behavior.

**Comment by ik9999** [comment]
I think i will add this service as a separate search engine soon:

https://www.valueserp.com/

Because im kinda tired of frequent changes in google search results page.

**Comment by ik9999** [comment]
I've added a new search engine here:

https://github.com/ik9999/popupsearch/releases/tag/v4.0.0
